plugins/informix-jdbc/src/main/java/com/navercorp/pinpoint/plugin/jdbc/informix/interceptor/InformixStatementCreateInterceptor.java

```diff
@@ -0,0 +1,117 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.pinpoint.plugin.jdbc.informix.interceptor;
+
+import com.navercorp.pinpoint.bootstrap.context.DatabaseInfo;
+import com.navercorp.pinpoint.bootstrap.context.SpanEventRecorder;
+import com.navercorp.pinpoint.bootstrap.context.Trace;
+import com.navercorp.pinpoint.bootstrap.context.TraceContext;
+import com.navercorp.pinpoint.bootstrap.interceptor.AroundInterceptor;
+import com.navercorp.pinpoint.bootstrap.logging.PLogger;
+import com.navercorp.pinpoint.bootstrap.logging.PLoggerFactory;
+import com.navercorp.pinpoint.bootstrap.plugin.jdbc.DatabaseInfoAccessor;
+import com.navercorp.pinpoint.bootstrap.plugin.jdbc.DefaultDatabaseInfo;
+import com.navercorp.pinpoint.bootstrap.plugin.jdbc.UnKnownDatabaseInfo;
+import com.navercorp.pinpoint.bootstrap.util.InterceptorUtils;
+
+import java.util.Arrays;
+import java.util.Properties;
+
+import com.navercorp.pinpoint.plugin.jdbc.informix.InformixConstants;
+import com.navercorp.pinpoint.plugin.jdbc.informix.interceptor.getter.DatabaseNameGetter;
+import com.navercorp.pinpoint.plugin.jdbc.informix.interceptor.getter.ConnectionInfoGetter;
+
+/**
+ * @author Guillermo Adrian Molina
+ */
+//@TargetMethods({
+//        @TargetMethod(name="createStatement"),
+//        @TargetMethod(name="createStatement", paramTypes={"int", "int"}),
+//        @TargetMethod(name="createStatement", paramTypes={"int", "int", "int"})
+//})
+public class InformixStatementCreateInterceptor implements AroundInterceptor {
+
+    private final PLogger logger = PLoggerFactory.getLogger(this.getClass());
+    private final boolean isDebug = logger.isDebugEnabled();
+
+    private final TraceContext traceContext;
+
+    public InformixStatementCreateInterceptor(TraceContext traceContext) {
+        this.traceContext = traceContext;
+    }
+
+    @Override
+    public void before(Object target, Object[] args) {
+        if (isDebug) {
+            logger.beforeInterceptor(target, args);
+        }
+    }
+
+    @Override
+    public void after(Object target, Object[] args, Object result, Throwable throwable) {
+        if (isDebug) {
+            logger.afterInterceptor(target, args, result, throwable);
+        }
+
+        if (InterceptorUtils.isThrowable(throwable)) {
+            return;
+        }
+        Trace trace = traceContext.currentTraceObject();
+        if (trace == null) {
+            return;
+        }
+        final DatabaseInfo databaseInfo = getDatabaseInfo(target);
+        if (result instanceof DatabaseInfoAccessor) {
+            ((DatabaseInfoAccessor) result)._$PINPOINT$_setDatabaseInfo(databaseInfo);
+        }
+        SpanEventRecorder recorder = trace.currentSpanEventRecorder();
+        recorder.recordServiceType(databaseInfo.getType());
+        recorder.recordEndPoint(databaseInfo.getMultipleHost());
+        recorder.recordDestinationId(databaseInfo.getDatabaseId());
```

Comment:
In general, the role of the StateCreateInterceptor is to attach the DatabaseInfo object for future use.
So, does not need to execute method of recording, because does not track at here. 
I wonder you have some reason for this, otherwise please remove it.

Reply:
This is related to the previous question. It is at this point that I can query the database info, that it's otherwise invalid. But I don't really fully understand pinpoint internals, so I there is another way of doing it, please let me know

Reply:
I checked it again. 
Honestly, I don't seem to need this.
If you look at other JDBC driver plugins, there is such a process. (They all do this in the Connection constructor.)
However, in this case, it is for augmenting the tracking data when creating a connection.

However, creating a Connection within the createStatement method or augmenting this data seems unusual, so I think this is unnecessary.
If you don't have this logic, I'm wondering if the look you're tracking will look different.

Reply:
I will check it out

Reply:
Ok, I've tested your proposal. You were right, It does work ok. I didn't understand you at first. I though you want me to kill the entire interceptor, not just the recording part. I show you prove that it still works:

![After](https://user-images.githubusercontent.com/16609390/78705055-2d506280-790d-11ea-8d2a-b1459ee47faf.png)